packages/backend/src/controllers/sqlRunnerController.ts

```diff
@@ -69,4 +73,72 @@ export class SqlRunnerController extends BaseController {
                 .getWarehouseFields(req.user!, projectUuid, tableName),
         };
     }
+
+    @Middlewares([
+        allowApiKeyAuthentication,
+        isAuthenticated,
+        unauthorisedInDemo,
+    ])
+    @SuccessResponse('200', 'Success')
+    @Post('/run')
+    @OperationId('runSql')
+    async runSql(
+        @Path() projectUuid: string,
+        @Body() body: SqlRunnerBody,
+        @Request() req: express.Request,
+    ): Promise<ApiJobScheduledResponse> {
+        this.setStatus(200);
+
+        return {
+            status: 'ok',
+            results: await this.services
+                .getProjectService()
+                .scheduleSqlJob(req.user!, projectUuid, body.sql),
+        };
+    }
+
+    /**
+     * Get results from a file stored locally
+     * @param fileId the fileId for the file
+     * @param req express request
+     */
+    @Middlewares([allowApiKeyAuthentication, isAuthenticated])
+    @SuccessResponse('200', 'Success')
+    @Get('results/{fileId}')
+    @OperationId('getLocalResults')
+    async getLocalResults(
+        @Path() fileId: string,
+        @Path() projectUuid: string,
+        @Request() req: express.Request,
+    ): Promise<any> {
+        this.setStatus(200);
+        this.setHeader('Content-Type', 'application/json');
+
+        const resultsFilePath = await this.services
+            .getProjectService()
+            .getResultsFile(req.user!, projectUuid, fileId);
+
+        // TODO set response headers
+        /*
+ //https://github.com/lukeautry/tsoa/issues/800
+       const stat: fs.Stats = await fs.promises.stat(filePath);
+
+      this.setStatus(200);
+      this.setHeader('Content-Type', mime.lookup(filePath));
+      this.setHeader('Content-Length', stat.size.toString());
+      // Removing this line will cause to not launch the download, just serve the file as it
+      this.setHeader('Content-Disposition', `attachment; filename=${fileName}`);
+      */
+        const mystream = fs.createReadStream(resultsFilePath);
+        const { res } = req;
+        if (res) {
+            mystream.pipe(res);
```

Comment:
we could also implement this endpoint outside of tsoa

Reply:
if we are going to keep this, I should probably return the readStream in `getResultsFile` instead of the path (I though tsoa should be able to read from a path like express.readFile)